includes/blocks.inc

```diff
@@ -74,23 +74,88 @@ function islandora_basic_collection_clear_object_count_cache() {
  * Outputs a JSON string to the buffer.
  */
 function islandora_basic_collection_object_count_callback() {
+  $count_method = variable_get('islandora_basic_collection_count_method', 'SPARQL');
+
+
   $cid = islandora_basic_collection_get_object_count_block_cache_id();
 
   if ($value = cache_get($cid, 'cache')) {
     $substitutions = $value->data;
   }
   else {
+
     $tuque = islandora_get_tuque_connection();
     $objects_query_array = islandora_basic_collection_get_query_info(array(
       'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
       'collection_listing' => TRUE,
       'all_objects' => TRUE,
     ));
     $collection_query_array = islandora_basic_collection_get_query_info(array(
-      'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
-      'collection_listing' => TRUE,
-      'all_objects' => FALSE,
-    ));
+    'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
+    'collection_listing' => TRUE,
+    'all_objects' => FALSE,
+  ));
+
+
+  // Setup Solr query for collection and object counts.
+  if ($count_method == "Solr") {
+    $all_cmodels = islandora_get_content_models();
+    $models_to_exclude = variable_get('islandora_basic_collection_object_count_listing_content_models_to_restrict', FALSE);
+    $cmodels_to_count = array();
+    foreach($all_cmodels AS $cmodel) {
+      if ($models_to_exclude[$cmodel['pid']] != '0') {
+        $cmodels_to_count[] = $cmodel['pid'];
+      }
+    }
+
+    // Collection count Solr query.
+    $collection_query = "RELS_EXT_hasModel_uri_mt:collection";
+    $qp = new IslandoraSolrQueryProcessor();
+    $qp->buildQuery($collection_query);
+    $qp->solrParams['fl'] = "PID";
+    $qp->solrLimit = 10000000;
+    $qp->executeQuery(FALSE);
+    try {
+      $results = $qp->islandoraSolrResult['response']['numFound'];
+    }
+    catch (Exception $e) {
+      watchdog_exception('Islandora Basic Collection', $e, 'Got an exception searching counting Collections.', array(), WATCHDOG_ERROR);
+      $results = array();
+    }
+    $collection_count = $results;
+    // Solr query for object count.
+    $object_query = "";
+    $i = 0;
+    foreach ($cmodels_to_count AS $model) {
+      if ($i == 0) {
+        $object_query = 'RELS_EXT_hasModel_uri_mt:"' . $model . '"';
+      }
+      else {
+        $object_query = $object_query . ' OR RELS_EXT_hasModel_uri_mt:"' . $model . '"';
```

Comment:
Should probably make use of the [configured model field](https://github.com/Islandora/islandora_solr_search/blob/77d9a268262e6dc94335fb8ac88f7e557fc0a514/includes/admin.inc#L660-L668), instead of hard-coding one.

Reply:
I did not know that was a thing! So I can do a `module_load_include('islandora_solr_search', 'inc', 'admin')` and create a variable referencing that field?

Reply:
Not... quite? It's whatever Drupal variable that it ends up getting stored in: https://github.com/Islandora/islandora_solr_search/blob/77d9a268262e6dc94335fb8ac88f7e557fc0a514/includes/admin.inc#L666

... Can just do the `variable_get()` thing (much the same as what the form does for its `#default_value`).